gocyclo: 2015-02-08 -> 0.4.0

vs49688 opened the pull request:
###### Motivation for this change

Updating.

We're pulling from a random fork, so switch to the actual upstream.

Also change to use buildGoModule instead of buildGoPackage.

Should supersede https://github.com/NixOS/nixpkgs/pull/145102.

@stackshadow I can add `Co-Authored-By:` if you'd like.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

stackshadow commented:
Would be cool, i didn't had the time yet to maintain my last PR for this package. Was a very busy month 😄

bobby285271 commented on pkgs/development/tools/gocyclo/default.nix:
```diff
@@ -1,23 +1,20 @@
+
   meta = with lib; {
     description = "Calculate cyclomatic complexities of functions in Go source code";
     homepage = "https://github.com/alecthomas/gocyclo";
```
```suggestion
    homepage = "https://github.com/fzipp/gocyclo";
```

vs49688 replied:
Oh fiddlesticks, I fixed that originally, but must've force-pushed over it...

vs49688 replied:
Fixed

vs49688 committed:
gocyclo: 2015-02-08 -> 0.4.0

We're pulling from a random fork, so switch to the actual upstream.

Also change to use buildGoModule instead of buildGoPackage.

Co-Authored-By: Martin <stackshadow@evilbrain.de>

bobby285271 committed:
Apply suggestions from code review

bobby285271 commented:
@ofborg build gocyclo

stackshadow commented:
![grafik](https://user-images.githubusercontent.com/7512804/152644309-34e32547-2d1b-4376-9dec-af9c42c35f47.png)

Okay ther is nothing changed according to the "old" cli-interface.

But the calulation is different 🤔 
![grafik](https://user-images.githubusercontent.com/7512804/152644420-56109011-0720-4613-ae6d-6198e5d4353a.png)
